libraries/abstractions/ble_hal/test/include/iot_test_ble_hal_common.h

```diff
@@ -203,164 +164,330 @@ typedef struct
+    {
+        BLEHALEventsInternals_t xEvent;
+        uint16_t usConnId;
+        uint32_t ulTransId;
+        BTBdaddr_t xBda;
+        uint16_t usAttrHandle;
+        uint16_t usOffset;
+    } BLETESTreadAttrCallback_t;
+
+    typedef struct
+    {
+        BLEHALEventsInternals_t xEvent;
+        uint16_t usConnId;
+        uint32_t ulTransId;
+        BTBdaddr_t xBda;
+        uint16_t usAttrHandle;
+        uint16_t usOffset;
+        size_t xLength;
+        bool bNeedRsp;
+        bool bIsPrep;
+        uint8_t ucValue[ bletestsSTRINGYFIED_UUID_SIZE ];
+    } BLETESTwriteAttrCallback_t;
+
+
+    typedef struct
+    {
+        BLEHALEventsInternals_t xEvent;
+        uint16_t usConnId;
+        BTStatus_t xStatus;
+    } BLETESTindicateCallback_t;
+
+    typedef struct
+    {
+        BLEHALEventsInternals_t xEvent;
+        BTStatus_t xStatus;
+        uint16_t usAttrHandle;
+    } BLETESTconfirmCallback_t;
+
+    typedef struct
+    {
+        BLEHALEventsInternals_t xEvent;
+        BTBdaddr_t xRemoteBdAddr;
+        BTBdname_t xRemoteBdName;
+        uint32_t ulCod;
+        BTSspVariant_t xPairingVariant;
+        uint32_t ulPassKey;
+    } BLETESTsspRequestCallback_t;
+
+    typedef struct
+    {
+        BLEHALEventsInternals_t xEvent;
+        BTStatus_t xStatus;
+        BTBdaddr_t xRemoteBdAddr;
+        BTSecurityLevel_t xSecurityLevel;
+        BTAuthFailureReason_t xReason;
+    } BLETESTPairingStateChangedCallback_t;
+
+    typedef struct
+    {
+        BLEHALEventsInternals_t xEvent;
+        uint16_t usConnId;
+        uint32_t ulTransId;
+        BTBdaddr_t xBda;
+        bool bExecWrite;
+    } BLETESTRequestExecWriteCallback_t;
+
+    typedef struct
+    {
+        BLEHALEventsInternals_t xEvent;
+        BTStatus_t xStatus;
+        uint16_t usConnId;
+        uint8_t ucServerIf;
+        BTBdaddr_t pxBda;
+        bool bConnected;
+    } BLETESTConnectionCallback_t;
+
+    typedef struct
+    {
+        BLEHALEventsInternals_t xEvent;
+        BTStatus_t xStatus;
+        BTBdaddr_t xRemoteBdAddr;
+        bool bIsBonded;
+    } BLETESTBondedCallback_t;
+
+
+    typedef enum
+    {
+        bletestATTR_SRVCA_SERVICE,
+        bletestATTR_SRVCA_CHAR_A,
+        bletestATTR_SRVCA_NUMBER
+    } bletestAttSrvA_t;
+
+    typedef enum
+    {
+        bletestATTR_SRVCB_SERVICE,
+        bletestATTR_SRVCB_CHAR_A,
+        bletestATTR_SRVCB_CHAR_B,
+        bletestATTR_SRVCB_CHAR_C,
+        bletestATTR_SRVCB_CHAR_D,
+        bletestATTR_SRVCB_CHAR_E,
+        bletestATTR_SRVCB_CCCD_E,
+        bletestATTR_SRVCB_CHAR_F,
+        bletestATTR_SRVCB_CCCD_F,
+        bletestATTR_SRVCB_CHARF_DESCR_A,
+        bletestATTR_SRVCB_CHARF_DESCR_B,
+        bletestATTR_SRVCB_CHARF_DESCR_C,
+        bletestATTR_SRVCB_CHARF_DESCR_D,
+        bletestATTR_INCLUDED_SERVICE,
+        bletestATTR_SRVCB_NUMBER
+    } bletestAttSrvB_t;
+
+    void pushToQueue( IotLink_t * pEventList );
+    void prvSetGetProperty( BTProperty_t * pxProperty,
```

Comment:
If these are private functions ( only scope is within a file ) then better make them static to avoid name collisions with other modules.